lib/oli_web/controllers/session_controller.ex

```diff
@@ -22,4 +25,15 @@ defmodule OliWeb.SessionController do
 
   defp session_data_to_delete(type),
     do: [String.to_atom("current_#{type}_id") | @shared_session_data_to_delete]
+
+  defp delete_cache_entry(conn, type) do
+    id =
+      conn.assigns
+      |> Map.get(String.to_existing_atom("current_#{type}"))
```

Comment:
To understand a little more... Here, are you getting an existing key in the assigns from the string you are passing? 🤔

Reply:
We are getting `:current_user` or `:current_author` from the assigns map in the connection `conn`